src/absenceDays/index.ts

```diff
@@ -1,24 +1,30 @@
 import BaseApi from '../baseApi';
 import responseHandler from '../utils/responseHandlers';
+import RequestParams from '../utils/requestParams';
 import { AbsenceDay } from './types';
 import { ReadParams } from '../utils/types';
 import { AxiosRequestConfig } from 'axios';
 
 export default class AbsenceDays extends BaseApi {
-  static resourceName = 'absencesDays';
+  private readonly resourceName = 'absencesDays';
 
-  public read(options: ReadParams = {}): Promise<AbsenceDay[]> {
-    const axiosConfig: AxiosRequestConfig = {
-      params: options,
-    };
-    const response = this._get<AbsenceDay[]>(`${AbsenceDays.resourceName}/read`, axiosConfig);
+  public read(requestParams?: RequestParams<AbsenceDay>): Promise<AbsenceDay[]> {
+    const params = requestParams ? requestParams.getParams() : {};
+    const response = this._get<AbsenceDay[]>(`${this.getResourceName()}/read`, { params });
     return responseHandler.requiredList(response);
   }
-  public readById(id: number, options: ReadParams = {}): Promise<AbsenceDay[]> {
-    const axiosConfig: AxiosRequestConfig = {
-      params: options,
-    };
-    const response = this._get<AbsenceDay[]>(`${AbsenceDays.resourceName}/read/${id}`, axiosConfig);
+
+  public readById(id: number, requestParams?: RequestParams<AbsenceDay>): Promise<AbsenceDay[]> {
+    const params = requestParams ? requestParams.getParams() : {};
+    const response = this._get<AbsenceDay[]>(`${this.getResourceName()}/read/${id}`, { params });
     return responseHandler.required(response);
   }
+
+  public getResourceName(): String {
```

Comment:
Would it make more sense to have it in Base and resourceName protected? Than we could also move getResourcePath to base as no it makes little sense to have it in each class.

Reply:
Yes is correct, but I think this should be done in another PR